### ISSUE TITLE
podman run fails inside of a Docker container in AWS Batch Job

### ISSUE THREAD

**Issue by AntonioAmore** [containers/podman#10722]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I launch an AWS batch job - a docker container, with podman inside. Podman successfully pulls a container, and dies on running  it with `invalid memory address or nil pointer dereference`. The issue appeared after podman v3.2.1 published.
**Steps to reproduce the issue:**

1.Create a Docker image, based on alpine with such endpoint: podman pulls a docker image, and runs it

2.Create AWS batch job, associated with the docker image from item 1

3.Launch the batch job, get `invalid memory address or nil pointer dereference` at job's logs

**Describe the results you received:**
A stack trace I collected from logs of the cloud job:
```
panic: runtime error: invalid memory address or nil pointer dereference,
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x560d50411b28],
github.com/containers/podman/v3/pkg/cgroups.getAvailableControllers(0xc000376870, 0xc0002a9000, 0x0, 0x15, 0xc0005f1038, 0x15, 0x1),
goroutine 1 [running]:,
pkg/cgroups/cgroups.go:377 +0xfa,
libpod/oci_conmon_linux.go:1401 +0x26e,
pkg/cgroups/cgroups.go:171 +0x688,
github.com/containers/podman/v3/libpod.(*ConmonOCIRuntime).moveConmonToCgroupAndSignal(0xc0005c2dc0, 0xc0005c2280, 0xc00054c6e0, 0xc00051f240, 0x0, 0x0),
github.com/containers/podman/v3/pkg/cgroups.New(0xc0005f1038, 0x15, 0xc000254df0, 0xc0005f1038, 0x15, 0xc000000180),
github.com/containers/podman/v3/libpod.(*ConmonOCIRuntime).createOCIContainer(0xc0005c2dc0, 0xc0005c2280, 0x0, 0x0, 0x0),
github.com/containers/podman/v3/libpod.(*Container).init(0xc0005c2280, 0x560d51639988, 0xc0001ac030, 0x0, 0x0, 0x0),
github.com/containers/podman/v3/libpod.(*Container).prepareToStart(0xc0005c2280, 0x560d51639988, 0xc0001ac030, 0x7f76f3b50501, 0x0, 0x0),
libpod/oci_conmon_linux.go:1157 +0x1305,
libpod/oci_conmon_linux.go:252 +0xb7,
github.com/containers/podman/v3/libpod.(*ConmonOCIRuntime).CreateContainer(0xc0005c2dc0, 0xc0005c2280, 0x0, 0x0, 0x0),
libpod/container_internal.go:1070 +0x1ae,
libpod/container_api.go:114 +0x9f,
github.com/containers/podman/v3/pkg/domain/infra/abi/terminal.StartAttachCtr(0x560d51639988, 0xc0001ac030, 0xc0005c2280, 0xc000134008, 0xc000134010, 0xc000134000, 0x560d50fac9e7, 0xd, 0x101, 0x0, ...),
pkg/domain/infra/abi/terminal/terminal_linux.go:91 +0x3d3,
libpod/container_internal.go:830 +0x315,
github.com/containers/podman/v3/libpod.(*Container).StartAndAttach(0xc0005c2280, 0x560d51639988, 0xc0001ac030, 0xc0005da930, 0x560d50fac9e7, 0xd, 0xc00004e420, 0xc0004da601, 0x0, 0x0, ...),
github.com/containers/podman/v3/pkg/domain/infra/abi.(*ContainerEngine).ContainerRun(0xc0000104f8, 0x560d51639988, 0xc0001ac030, 0x0, 0x0, 0x0, 0x560d50fac9e7, 0xd, 0xc000134010, 0xc000134000, ...),
github.com/spf13/cobra.(*Command).ExecuteC(0x560d52010320, 0xc000132020, 0x560d513be7e0, 0x560d520d07c8),
vendor/github.com/spf13/cobra/command.go:960 +0x375,
cmd/podman/containers/run.go:188 +0x5b5,
github.com/spf13/cobra.(*Command).execute(0x560d51ffaa20, 0xc000138020, 0x20, 0x22, 0x560d51ffaa20, 0xc000138020),
vendor/github.com/spf13/cobra/command.go:852 +0x472,
pkg/domain/infra/abi/containers.go:898 +0x37f,
github.com/containers/podman/v3/cmd/podman/containers.run(0x560d51ffaa20, 0xc000520800, 0x9, 0x20, 0x0, 0x0),
vendor/github.com/spf13/cobra/command.go:890,
github.com/spf13/cobra.(*Command).Execute(...),
main.Execute(),
vendor/github.com/spf13/cobra/command.go:897,
main.main(),
cmd/podman/root.go:90 +0xe9,
github.com/spf13/cobra.(*Command).ExecuteContext(...),
```

**Describe the results you expected:**
podman v.3.2.1 runs containers successfully inside docker container as an AWS BatchJob

**Additional information you deem important (e.g. issue happens only occasionally):**
v.3.1.2 and previous versions worked well for all environments I need.. All podman versions in that configuration (podman inside docker) runs perfectly on my local machine.

**Output of `podman version`:**

```
Version:      3.2.1,
API Version:  3.2.1,
Go Version:   go1.16.5,
Git Commit:   d183f0610827ab235b01ae4c6f8c8f2858d50284,
Built:        Mon Jun 14 17:24:45 2021,
OS/Arch:      linux/amd64,
```

**Output of `podman info --debug`:**

```
panic: runtime error: invalid memory address or nil pointer dereference
Then the trace stack I provided above
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
apk list podman
podman-3.2.1-r0 x86_64 {podman} (Apache-2.0) [installed]
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS Batch Job, Docker container from Amazon ECR, based on Alpine Linux 3.13

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
could it be fixed with fc9868e2247d2eb651586bd4b3e9a4ece3562fa0 ?

**Comment by giuseppe** [comment]
yes, I think that is the correct fix.  Closing the issue
